reco_analysis/reco_analysis/chatbot/chatbot.py

```diff
@@ -250,6 +280,9 @@ def receive(self, message: str) -> None:
         # Save the user input to the conversation memory
         self.memory.add_message(HumanMessage(content=message, name=self.human_role))
 
+        # Detect end of conversation if the role is Doctor
+        self.detect_and_handle_end() if self.role == 'Doctor' else None
```

Comment:
seeing that this isn't used in the streamlit_app currently, maybe we should parameterize it (add as a parameter in DialogueAgent's __init__), so that we turn it on only during synthetic transcript generation, but turn it off in the UI (to save costs). what do you think?

Reply:
Agreed. I'll parametrize this.